style-guide: organize content and update sections; *: fix Markdownlint errors (#12148)
tldr-pages/tldr: contributing-guides/style-guide.md

> More information: <https://learn.microsoft.com/powershell/module/microsoft.powershell.management/clear-recyclebin>.
```

> ## <a name="pages"></a>Pages

Author: From the HTML spec of 15 February 2024, section [7.4.6.4 Scrolling to a fragment](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scrolling-to-a-fragment):
> To find a potential indicated element given a [Document](https://html.spec.whatwg.org/multipage/dom.html#document) document and a string fragment, run these steps:
> 
> If there is an element [in the document tree](https://dom.spec.whatwg.org/#in-a-document-tree) whose [root](https://dom.spec.whatwg.org/#concept-tree-root) is document and that has an [ID](https://dom.spec.whatwg.org/#concept-id) equal to fragment, then return the first such element in [tree order](https://dom.spec.whatwg.org/#concept-tree-order).

>
> If there is an [a](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) element [in the document tree](https://dom.spec.whatwg.org/#in-a-document-tree) whose [root](https://dom.spec.whatwg.org/#concept-tree-root) is document that has a [name](https://html.spec.whatwg.org/multipage/obsolete.html#attr-a-name) attribute whose value is equal to fragment, then return the first such element in [tree order](https://dom.spec.whatwg.org/#concept-tree-order).
>
> Return null.

So either would work, but I think it's better to use `id`

> ## <a name="pages"></a>Pages

Reviewer: I have reverted the HTML tag addition to titles and added the index reference to the markdown headings directly.

### Platform differences

If you are afraid the commands may differ between platforms or operating systems (e.g. Windows vs macOS), most [tldr pages clients](https://github.com/tldr-pages/tldr/wiki/tldr-pages-clients) will choose the most suitable version of the command.

> In this case, the information of the Windows version of `cd` (stored in `pages/windows/cd.md`) will be displayed by default to Windows users, and a generic/common version (stored in `pages/common/cd.md`) will be displayed for Linux, macOS, and other platforms.

Author: I didn't edit this text, I just separated it from the "Imperative Mood" section because it didn't fit there, it almost looked like a random text in the middle of the section.

> In this case, the information of the Windows version of `cd` (stored in `pages/windows/cd.md`) will be displayed by default to Windows users, and a generic/common version (stored in `pages/common/cd.md`) will be displayed for Linux, macOS, and other platforms.

Reviewer 2: Okay, I see. But considering your PR in general the diff is quite extensive, is it just table of content headings? The diff has gone a bit mad.

> In this case, the information of the Windows version of `cd` (stored in `pages/windows/cd.md`) will be displayed by default to Windows users, and a generic/common version (stored in `pages/common/cd.md`) will be displayed for Linux, macOS, and other platforms.

Reviewer: I have currently rebased this PR to accommodate the recent changes to the style guide, I will provide a detailed changelog once my changes are complete.

- Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).
- Compression algorithms, e.g. `zip`, `7z`, `xz`.

## Heading

### More information links

- On the `More information` link line, we prefer linking to the author's provided documentation of the command line reference or the man page. When not available, use <https://manned.org> as the default fallback for all platforms
(except `osx` and BSD platforms other than FreeBSD).
Alternatively, you can link to the author's website or a tutorial page if the command doesn't have a documentation page.

- For `osx`: Apple distributes the built-in man pages [in Xcode](https://developer.apple.com/documentation/os/reading_unix_manual_pages).
For commands documented there, we recommend using <https://keith.github.io/xcode-man-pages/>, an HTML export of all Apple's man pages bundled with Xcode.

> [!IMPORTANT]
> All links must be enclosed inside angular brackets (`<` and `>`).

> - We prefer translations to use the more information link of the English page by default.

Reviewer: How about this?

```suggestion
- We suggest using the more information link of the English page for translations.
```
(or)
```suggestion
- It is suggested to use the more information link of the English page for translations.
```

We don't generally support showing localized versions of the pages in a majority of cases as the translations a lot of times are out of sync with the English page.

> - We prefer translations to use the more information link of the English page by default.

Reviewer 3: I'd personally prefer the first version.

- Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).
- Compression algorithms, e.g. `zip`, `7z`, `xz`.

## Heading

### More information links

- On the `More information` link line, we prefer linking to the author's provided documentation of the command line reference or the man page. When not available, use <https://manned.org> as the default fallback for all platforms
(except `osx` and BSD platforms other than FreeBSD).
Alternatively, you can link to the author's website or a tutorial page if the command doesn't have a documentation page.

- For `osx`: Apple distributes the built-in man pages [in Xcode](https://developer.apple.com/documentation/os/reading_unix_manual_pages).
For commands documented there, we recommend using <https://keith.github.io/xcode-man-pages/>, an HTML export of all Apple's man pages bundled with Xcode.

> [!IMPORTANT]
> All links must be enclosed inside angular brackets (`<` and `>`).

> - We prefer translations to use the more information link of the English page by default.

Author: > We don't generally support showing localized versions of the pages in a majority of cases as the translations a lot of times are out of sync with the English page.

My understanding: it's because the links can be outdated if they change, right?

If so, I agree this can be a problem. I prefer "It is suggested to", but I support making the argument clear in the style guide itself:

```suggestion
- It is suggested to use a more information link with English content in both translations and English pages. That's because the links can eventually change, but the translations are often out of sync with the English pages.
```

I've also thought that a link update can eventually come from a translation and propagate to the others using our script, so I reworded the first sentence, what do you think?

> [!NOTE]  
> In cases where the character isn't present in the translated word, you can highlight the option before/next to the equivalent word or you can add the English work beside the translation inside a bracket.
> For example, `E[x]tract` in English maybe translated into `[x] ekstrak` or `ekstrak [x]` or `ekstrak (E[x]tract)` in Indonesian.

## Example commands

### Option syntax

- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with [mnemonics](#short-option-mnemonics) or both inside a placeholder.
- For highlighting both long and short options in commands (instead of using mnemonics), combine them within a placeholder i.e. `{{-o|--output}}`.
- For user-friendliness, use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms) for pages in `common` directory.

### Placeholder syntax

User-provided values should use the `{{placeholder}}` syntax
in order to allow `tldr` clients to highlight them.

> [!TIP]

> > It is suggest to enclose placeholders accepting strings as input within quotes. i.e. Use `"{{placeholder}}"` instead of `{{"placeholder"}}`.

Author: ```suggestion
> It is suggested to enclose placeholders accepting strings as input within quotes. i.e. Use `"{{placeholder}}"` instead of `{{"placeholder"}}`.
```
IMO here we can make it a bit open for edge cases.

  but has to start at the root of the filesystem,
  prefix it with a slash,
  such as `get {{/path/to/remote_file}}`.
- In case of a possible reference both to a file or a directory,
  use `{{path/to/file_or_directory}}`.

> [!NOTE]  
> If the command is specific to Windows, use backslashes (`\`) instead, such as `{{path\to\file_or_directory}}`. Drive letters such as `C:` are optional unless the command input requires an absolute path
> or specific drive letter range, such as `cd /d {{C}}:{{path\to\directory}}`.

#### Extensions

- If a particular extension is expected for the file, append it.
  For example, `unrar x {{path/to/compressed.rar}}`.
- In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
  For instance, in `find.md`'s example "Find files by extension" (`find {{path/to/root}} -name '{{*.ext}}'`)
  using `{{*.ext}}` explains the command without being unnecessarily specific;
  while in `wc -l {{path/to/file}}` using `{{path/to/file}}` (without extension) is sufficient.

#### Grouping placeholders

- If a command can take 0 or more arguments of the same kind, use an ellipsis: `{{placeholder1 placeholder2 ...}}`.

>   For instance, if multiple paths are expected `{{path/to/directory1 path/to/directory2 ...}}` can be used.

Author: ```suggestion
  For instance, if multiple paths are expected `{{path/to/directory1 path/to/directory2 ...}}` can be used.
```

```suggestion
  For instance, if multiple paths are expected, use `{{path/to/directory1 path/to/directory2 ...}}`.
```
